src/Capgemini.PowerApps.PackageDeployerTemplate/Adapters/CrmServiceAdapter.cs

```diff
@@ -313,6 +315,72 @@ public TResponse Execute<TResponse>(OrganizationRequest request, string username
             return (TResponse)this.Execute(request);
         }
 
+        /// <inheritdoc/>
+        public bool HasStartedSolutionHistoryRecords(ILogger logger)
+        {
+            var retryPolicy = Policy
+                .HandleResult<bool>(hasActiveRecords => !hasActiveRecords)
+                .WaitAndRetryForever(
+                    sleepDurationProvider: retryAttempt => TimeSpan.FromSeconds(20),
+                    onRetry: (result, timeSpan) =>
+                    {
+                        logger.LogInformation($"Active records still present within Solution History.  Waiting for {timeSpan.TotalSeconds} seconds before retrying...");
+                    });
+
+            var result = retryPolicy.Execute(() =>
+            {
+                var entityCollection = this.crmSvc.RetrieveMultiple(new QueryExpression()
+                {
+                    EntityName = Constants.SolutionHistory.LogicalName,
+                    ColumnSet = new ColumnSet(
+                    Constants.SolutionHistory.Fields.SolutionHistoryId,
+                    Constants.SolutionHistory.Fields.Name,
+                    Constants.SolutionHistory.Fields.Status),
+                    Criteria =
+                    {
+                        Conditions =
+                        {
+                            new ConditionExpression(Constants.SolutionHistory.Fields.Status, ConditionOperator.Equal, Constants.SolutionHistory.Statuses.Started),
+                        },
+                    },
+                });
```

Comment:
`QueryByAttribute` would be more concise here.

Reply:
@ewingjm replace `QueryExpression` with `QueryByAttribute`